### PR TITLE
Doxygen Improvements

### LINKED ISSUE #29: Documentation: create doxygen code reference for releases

### PR DESCRIPTION
**Pull request by vvolkl**
BEGINRELEASENOTES
-  Improvements  to doxygen configuration 

ENDRELEASENOTES

A preview of the current doxygen page right now is here: https://vvolkl.github.io/EDM4HEP/doc/latest/index.html
I'll set up an automated deployment next.

Fixes #29

### CODE REVIEW

File: doc/DoxygenLayout.xml [hunk 1]
```diff
@@ -0,0 +1,196 @@
+<doxygenlayout version="1.0">
+  <!-- Generated by doxygen 1.8.11 -->
+  <!-- Navigation index tabs for HTML output -->
+  <navindex>
+    <tab type="user" visible="yes" title="EDM4hep Home Page" url="https://cern.ch/edm4hep"/>
```
Comment (andresailer) [inline doc/DoxygenLayout.xml]: This website doesn't exist, maybe pointing to edm4hep.gihub.io would be better?

**Comment by vvolkl** [comment]
Thanks for the comments. Yes, regarding the main page, I noted it here: https://github.com/HSF/EDM4hep/issues/29 I'll probably just remove the table and see if the image renders correctly if I use html.

File: doc/DoxygenLayout.xml (hunk 1 above)
Comment (vvolkl) [reply doc/DoxygenLayout.xml]: Hmmm, I did create `cern.ch/EDM4hep`, and redirected it to the gh-page for now,  but something seems to have gone wrong with the capitalization. I'll look into it.

**Comment by andresailer** [comment]
The html tags for the diagram are there:
```
<object type="image/svg+xml" data="doc/edm4hep_diagram.svg"></object>
```
But the relative path doesn't work from where the index.html lives

**Comment by andresailer** [comment]
The table is not well-formed, there is an additional "Vertex" entry is the second to last line
I think that gives you problems in the html.

```

| | | | |
| ------------- | ------------- | ----- | --- | 
| [Vector3f ](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L9) | [Vector3d ](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L23) | [Vector2i ](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L38) | [TrackState ](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L50) |
| [SimCalorimeterHit](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L201) | [RawCalorimeterHit](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L212) |[ CalorimeterHit ](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L222) | [Cluster](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L247) | 
| [MCParticle](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L84)      | [Vertex](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L318) | [ParticleID ](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L234) | [ReconstructedParticle](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L335) | 
| [SimTrackerHit](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L156) | [TPCHit](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L288) | [TrackerHit](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L270) | [Track](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L300) | [Vertex](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L318) |
| [MCRecoParticleAssociation](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L362) | [MCRecoCaloAssociation](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L370) | [MCRecoTrackerAssociation](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L370) | [ObjectID](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L71) |
```

**Comment by andresailer** [comment]
Emacsed:
```
|                                                                                           |                                                                                       |                                                                                          |                                                                                       |                                                                        |
| -------------                                                                             | -------------                                                                         | -----                                                                                    | ---                                                                                   |                                                                        |
| [Vector3f ](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L9)                   | [Vector3d ](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L23)              | [Vector2i ](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L38)                 | [TrackState ](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L50)            |                                                                        |
| [SimCalorimeterHit](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L201)         | [RawCalorimeterHit](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L212)     | [ CalorimeterHit ](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L222)         | [Cluster](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L247)               |                                                                        |
| [MCParticle](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L84)                 | [Vertex](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L318)                | [ParticleID ](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L234)              | [ReconstructedParticle](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L335) |                                                                        |
| [SimTrackerHit](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L156)             | [TPCHit](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L288)                | [TrackerHit](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L270)               | [Track](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L300)                 | [Vertex](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L318) |
| [MCRecoParticleAssociation](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L362) | [MCRecoCaloAssociation](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L370) | [MCRecoTrackerAssociation](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L370) | [ObjectID](https://github.com/HSF/EDM4hep/blob/master/edm4hep.yaml#L71)               |                                                                        |


```

File: doc/DoxygenLayout.xml (hunk 1 above)
Comment (andresailer) [reply doc/DoxygenLayout.xml]: Right, I get 403 (forbidden), not 404 (not found)

Comment (vvolkl) [reply doc/DoxygenLayout.xml]: Fixed. (Redirected to my fork for now, since this repository is not deployed)

**Comment by vvolkl** [comment]
Many thanks, good catch! It renders correctly now, save for a "/>" after the svg. I don't have permissions to enable the deployment for this repo. This is probably a good thing to do after transferring the repo to the key4hep organization

**Comment by andresailer** [comment]
Needs to be adapted to the new home of EDM4hep in the key4hep project.

**Comment by andresailer** [comment]
It is good to merge now, and then you enable the github pages deployment?

**Comment by vvolkl** [comment]
Hi, Andre, I think I'll do that in this PR. I haven't found a good template script to deploy doxygen to gh-pages via travis, but in principle it is fairly straightforward, so it shouldn't take too long.

**Comment by andresailer** [comment]
Ok, great

**Comment by vvolkl** [comment]
Fixes #40

**Comment by vvolkl** [comment]
This should now update the doxygen page every time a pull request is merged. Also, since we don't have a separate edm4hep homepage, I changed the link on the main page to the key4hep, which I think should serve as the central documentation.

**Comment by vvolkl** [comment]
It seems I messed up my git configuration while testing the deployment, will rebase once more with the proper git user.name

**Pull request merged**